lib/gui/modules/image-writer.js

```diff
@@ -102,12 +113,32 @@ imageWriter.service('ImageWriterService', function($q, $rootScope) {
         flashState.setProgressState(state);
       });
 
-    }).then(flashState.unsetFlashingFlag).catch((error) => {
+    }).then(flashState.unsetFlashingFlag).then(() => {
+      if (flashState.wasLastFlashCancelled()) {
+        AnalyticsService.logEvent('Elevation cancelled', analyticsData);
+      } else {
+        AnalyticsService.logEvent('Done', analyticsData);
+      }
+    }).catch((error) => {
       flashState.unsetFlashingFlag({
         errorCode: error.code
       });
 
+      if (error.code === 'EVALIDATION') {
+        AnalyticsService.logEvent('Validation error', analyticsData);
+      } else if (error.code === 'EUNPLUGGED') {
+        AnalyticsService.logEvent('Drive unplugged', analyticsData);
+      } else if (error.code === 'EIO') {
+        AnalyticsService.logEvent('Input/output error', analyticsData);
+      } else if (error.code === 'ENOSPC') {
+        AnalyticsService.logEvent('Out of space', analyticsData);
+      } else {
+        AnalyticsService.logEvent('Flash error', analyticsData);
```

Comment:
If this is catching cases where we haven't checked a matching `error.code`, maybe it'd be worth this logEvent reporting the `error.code` that triggered it?

EDIT: Ah, looks like the previous code in `lib/gui/pages/main/controllers/flash.js` actually used to log the entire `error` object :)  Probably worth doing the same still?